src/CometFlexVoting.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.10;
+
+import {Comet} from "comet/Comet.sol";
+import {CometConfiguration} from "comet/CometConfiguration.sol";
+import {Checkpoints} from "@openzeppelin/contracts/utils/Checkpoints.sol";
+
+import {IVotingToken} from "src/interfaces/IVotingToken.sol";
+import {IFractionalGovernor} from "src/interfaces/IFractionalGovernor.sol";
+import {FlexVotingClient} from "src/FlexVotingClient.sol";
+
+// TODO add description
+contract CometFlexVoting is Comet, FlexVotingClient {
+  using Checkpoints for Checkpoints.History;
+
+  /// @dev Constructor.
+  /// @param _config The configuration struct for this Comet instance.
+  /// @param _governor The address of the flex-voting-compatible governance contract.
+  constructor(CometConfiguration.Configuration memory _config, address _governor)
+    Comet(_config)
+    FlexVotingClient(_governor)
+  {
+    selfDelegate();
+  }
+
+  /// @notice Returns the _user's current balance in storage.
+  function _rawBalanceOf(address account) internal view override returns (uint256) {
+    int104 _principal = userBasic[account].principal;
+    return _principal > 0 ? uint256(int256(_principal)) : 0;
```

Comment:
why is this conditional needed? since `uint256(int256(int104(0)))` equals 0. Also this should have a safe cast to ensure the int value fits in uint. RR
```suggestion
    return toUint256(int256(userBasic[account].principal));
```

where `toUint256` is something like the solmate or OZ safecast methods

Reply:
> why is this conditional needed?

Because `_principal` can be negative, e.g. if someone has borrowed the base asset

> Also this should have a safe cast to ensure the int value fits in uint

Why do we need a safe cast? If the value fits in `int104` and it's positive then we know it fits in `uint256`

Reply:
ah yes that all makes sense! though it seems we're not actually using the `_rawBalanceOf` methods anywhere, should they all be removed?

Reply:
oops somehow I missed that in my code search, resolving this! 😅